cabal-install/Distribution/Client/ProjectPlanOutput.hs

```diff
@@ -193,10 +193,13 @@ encodePlanAsJson distDirLayout elaboratedInstallPlan elaboratedSharedConfig =
             J.object [ "type" J..= J.String "repo-tar"
                      , "repo" J..= repoToJ repo
                      ]
-          RemoteSourceRepoPackage srcRepo _ ->
-            J.object [ "type" J..= J.String "source-repo"
-                     , "source-repo" J..= sourceRepoToJ srcRepo
-                     ]
+          RemoteSourceRepoPackage srcRepo local ->
+            J.object $
+              [ "type" J..= J.String "source-repo"
+              , "source-repo" J..= sourceRepoToJ srcRepo
+              ] ++
+              [ "path" J..= J.String path | path <- maybeToList local ]
```

Comment:
is this cloned or `sdist`d path?

Reply:
I think the field should be `local-tarball`, and it should be also added for `RemoteTarballPackage`.

Reply:
It's the path to the locally cloned repository. It's got nothing to do with tarballs, here's an example:

```
  "pkg-src": {
    "type": "source-repo",
    "source-repo": {
      "type": "git",
      "location": "../../../../",
      "tag": "HEAD",
      "subdir": "tests/exelib"
    },
    "path": "/home/dxld/share/dev/hs/cabal-helper/tests/src-repo/dist-newstyle/src/.-6e043ec6becad825/tests/exelib"
  },
```

Reply:
For tarballs what I'd want is the directory they get unpacked into, which I'm not so sure is available at this stage. I can at least add the path to the tarball though if you like.